src/hotspot/share/gc/g1/g1HeapRegionAttr.hpp

```diff
@@ -103,6 +103,7 @@ struct G1HeapRegionAttr {
   bool is_young() const                { return type() == Young; }
   bool is_old() const                  { return type() == Old; }
   bool is_optional() const             { return type() == Optional; }
+  bool needs_bot_update() const        { return is_old(); }
```

Comment:
Not sure if that predicate needs to be here, I'd probably just add a method to `G1PLABAllocator`. But it is fine to me.